bitcoin-core-dev-tech/2023-09/p2p-working-session.md

```diff
@@ -0,0 +1,204 @@
+  - I took from my addrman and used an ASMap and saw that 8 or 9 autonomous systems is what I saw. Saw that the random selection is always more than 6 autonomous systems.
+  - is this worth spending time on. An attacker that only has access to Google cloud or AWS, how easy would it be to do?
+  - you should run it when a different /16 with an empty addrman. You already have data and if you take out that restriction.
+  - this will influence on the connections of the network (as it grows). ASs are one approximation for control. How good we don't know, but better than slash 16s. AWS probably has double digits AS numbers and thousands of IP ranges.
+  - botnets have different looking IPs in many different ranges. Scattered.
+  - If you could do a botnet and the network grew and then an attacker could position themselves to get more connections (Virtu presentation described effects of scaled up network with constant shares between ASs)
+  - In the case of 100x scale-up, random selection might be better than ASMap selection if [virtu's research](https://cryptic.to/advancing-bitcoin-2023/) is a concern. /16s make no difference.
+- AWS has multiple AS numbers, this softens the effect virtu described
+- we could choose to merge all of AWS into a virtual AS. The question is where do we get the data from. I don't know where the place to have this sort of discussion apart from here. Does it belong on an issue or the mailing list?
+- before it is included in a release, it makes sense to post it on the mailing list.
+
+## Broadcasting out all txs through Tor (Vasil)
+
+- The idea is to broadcast local transactions from the wallet or RPC - right now we broadcast to all peers, which makes it easy to correlate txs to IP addresses.
+- Only broadcast to Tor peers if possible. Send to someone random and disconnect. If I only tell Tor peers and a safety net to tell if it is propagated and if not broadcast.
+- Can see the timing and where txs originate from.
+- The trickling mechanism adds randomness and doesn't obscure things from an attacker that sees everything
+- Inspired by [Linking Lion](https://b10c.me/observations/05-inbound-connection-flooder-down/)
+- Problem is the change and where to put that.
+- If we send it to a few peers over Tor and some of them are spies, they have a privileged position to see. If they come to our mempool and ask and they can confirm if it is yours. Then if some asks and then we pretend we don't have it.
+  - you are changing the behavior of what we allow in the mempool.
+  - If some sends an INV or sends GETDATA - parent and child issues also complicate things. Convinced I can hide it.
+- Now I'm leaning the other way, which is not to put it in the mempool. But where? Need to put it on disk. In the mempool there are unbroadcasted txs. The code is already there. If it doesn't come back, you need to put it somewhere.
```

Comment:
```suggestion
- Now I'm leaning the other way, which is to broadcast without putting it in the mempool. Need to store it on disk too before it as been received by the network, so that it does not get lost if the the node restarts. In the mempool there are unbroadcasted txs. The code is already there. If it doesn't come back, you need to store it safely for retry later.
```